PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -154,8 +143,126 @@ ParseIasImage (
     return EFI_UNSUPPORTED;
   }
 
+  return Status;
+}
+
+/**
+  Parse PODS image
+
+  This function will parse PODS image and get every file blobs info, Especially
+  get the command line file and ELF/kernel file for boot.
+
+  @param[in]      BootOption    Current boot option
+  @param[in, out] LoadedImage   Loaded Image information.
+
+  @retval  RETURN_SUCCESS       Parse IAS image successfully
+  @retval  Others               There is error when parsing IAS image.
+**/
+EFI_STATUS
+ParseContainerImage (
+  IN     OS_BOOT_OPTION      *BootOption,
+  IN OUT LOADED_IMAGE        *LoadedImage
+  )
+{
+  EFI_STATUS                  Status;
+  CONTAINER_HDR              *ContainerHdr;
+  UINT64                      ComponentName;
+  LOADER_COMPRESSED_HEADER   *LzHdr;
+  IMAGE_DATA                  File[MAX_IAS_SUB_IMAGE];
+  UINT8                       Index;
+
+  ContainerHdr = (CONTAINER_HDR  *)LoadedImage->IasImage.Addr;
```

Comment:
Should we check the container signature here ?    Only handle signature with 'BOOT' ?